vm/src/builtins/list.rs

```diff
@@ -397,12 +397,8 @@ impl PyList {
     }
 
     #[pyslot]
-    fn tp_new(
-        cls: PyTypeRef,
-        _iterable: OptionalArg<PyObjectRef>,
-        vm: &VirtualMachine,
-    ) -> PyResult<PyRef<Self>> {
-        PyList::default().into_ref_with_type(vm, cls)
+    fn tp_new(cls: PyTypeRef, _iterable: FuncArgs, vm: &VirtualMachine) -> PyResult {
```

Comment:
```suggestion
    fn tp_new(cls: PyTypeRef, _args: FuncArgs, vm: &VirtualMachine) -> PyResult {
```
This change enhances compatibility :+1:, but the argument name doesn't fit to its actual value anymore